io/jvm/src/main/scala/org/scalajs/io/AtomicFileOutputStream.scala

```diff
@@ -43,8 +44,17 @@ private[io] class AtomicFileOutputStream private (
     if (!tmpFile.renameTo(baseFile)) {
       // Renaming failed. Fallback to copy
       try {
-        IO.copyTo(FileVirtualBinaryFile(tmpFile),
-            WritableFileVirtualBinaryFile(baseFile))
+        try {
+          // Try atomic move copy.
+          Files.copy(tmpFile.toPath, baseFile.toPath,
```

Comment:
There doesn't appear to be such a thing as atomic *copy*. See [here](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#copy-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-):

> Copying a file is not an atomic operation.

What you want is probably an atomic [`move`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#move-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-), which, among other things, says:

> `ATOMIC_MOVE`: The move is performed as an atomic file system operation and all other options are ignored. [...]

so `REPLACE_EXISTING` would be redundant, and

> If the move cannot be performed as an atomic file system operation then `AtomicMoveNotSupportedException` is thrown.

Note that `AtomicMoveNotSupportedException` is not a subtype of `UnsupportedOperationException` so the catch below needs to be adapted.

Reply:
Also, this specific change should probably be in its own commit. It may have been prompted by the removal of `IO.copyTo`, but using an actual `ATOMIC_MOVE` will improve the observable quality of this implementation, so it doesn't fit in a commit that says "Remove most the API that is unused".